### DISCUSSION TITLE
In the CROSS_THREAD scenario where spanA asynchronously calls spanB, if spanA is ignored, can spanB also be automatically ignored?

### DISCUSSION THREAD

**Discussion by gzlicanyi** [apache/skywalking#12161] [Q&A] [answered]
In the CROSS_THREAD scenario where spanA asynchronously calls spanB, if spanA is ignored, can spanB also be automatically ignored?

**Comment by wu-sheng** [comment]
How could you be possible to ignore a span only? SkyWalking sampling policy is segment oriented. 

A span is created before `continued(CROSS_THREAD)`. They are irrelevant. Same story for service A calling service B. Sampling are controlled at both ends.

**Reply by gzlicanyi** [reply]
In the same service, I want to ignore some unimportant interfaces with large traffic (EntrySpanA). If EntrySpanA is ignored, its synchronous sub-spans will also be ignored, but asynchronous sub-spans won't be.

**Reply by gzlicanyi** [reply]
I understand that the reason is different TracerContexts for asynchronous operations. So I want to ask if it is possible to change the TracerContext of a sub-span to IgnoredTracerContext during a continued operation

**Reply by wu-sheng** [reply]
Why don't you put the operation name of your first span in your async process into the ignore filter?

**Reply by gzlicanyi** [reply]
If the operation name is generated by a third-party plugin, such as [Spring Cloud Gateway](https://github.com/apache/skywalking-java/blob/1a010470076ba7f206f4fd6559d04136bea73417/apm-sniffer/optional-plugins/optional-spring-plugins/optional-spring-cloud/gateway-2.0.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/cloud/gateway/v20x/NettyRoutingFilterInterceptor.java#L40), is there a more elegant way to handle this?

**Reply by wu-sheng** [reply]
If this is Spring Cloud gateway, I think you can't do that easily.

**Reply by gzlicanyi** [reply]
Can I modify it such that if ContextSnapshot isValid() == false when the createLocalSpan() method is called, I set the operationName to null, causing the ContextManager to create an IgnoredTracerContext?

**Reply by wu-sheng** [reply]
At least you can't do that right now.

**Reply by gzlicanyi** [reply]
Why? I can submit a PR.

**Reply by wu-sheng** [reply]
I mean, `set the operationName to null` to trigger IgnoredTracerContext is not a recommend way. And this will trigger different behavior with other plugins. 

If you want to do this, we need to add a mechanism like `forceSampling` or `forceIgnoring` mechanism.

**Comment by wu-sheng** [comment, marked as answer]
Implemented through https://github.com/apache/skywalking-java/pull/689
